src/me/mrCookieSlime/ExoticGarden/ExoticGarden.java

```diff
@@ -843,24 +841,26 @@ public static ItemStack harvestPlant(Block block) {
 							Block plant;
 							if (BlockStorage.check(block.getRelative(BlockFace.DOWN)) == null) {
 								plant = block;
-								BlockStorage.retrieve(block.getRelative(BlockFace.UP));
+								BlockStorage.clearBlockInfo(block.getRelative(BlockFace.UP));
 								block.getWorld().playEffect(block.getRelative(BlockFace.UP).getLocation(), Effect.STEP_SOUND, Material.OAK_LEAVES);
-								block.getRelative(BlockFace.UP).setType(Material.AIR);;
+								block.getRelative(BlockFace.UP).setType(Material.AIR);
 							}
 							else {
 								plant = block.getRelative(BlockFace.DOWN);
-								BlockStorage.retrieve(block);
+								BlockStorage.clearBlockInfo(block);
 								block.getWorld().playEffect(block.getLocation(), Effect.STEP_SOUND, Material.OAK_LEAVES);
-								block.setType(Material.AIR);;
+								block.setType(Material.AIR);
 							}
 							plant.setType(Material.OAK_SAPLING);
 							itemstack = berry.getItem();
+							BlockStorage._integrated_removeBlockInfo(plant.getLocation(), false);
```

Comment:
Is this safe to use here?

Reply:
I didn't have issues with it while running a server.
The problem here is that we are adding the new block data at the same time as removing the old one *(like `ID` berry -> `ID_BUSH` upon harvesting)*, which is not normally possible. Methods like `clearBlockInfo(loc)` add block into the ticker task, so upon removing we'll have two entries in the memory for the same location, which causes a lot of problems *(issues #61 and #88)*. This forces Slimefun to remove the data immediately.

Reply:
Alright, thanks for the clarification